build/nuget-release.yml

```diff
@@ -120,7 +120,7 @@ stages:
 
   - stage: SelfContainingIntegrationTests
     displayName: Self-Containing Integration Tests
-    dependsOn: Build
+    dependsOn: DockerIntegrationTests
```

Comment:
Why are we changing this? They don't depend on each other, do they 🤔

Reply:
Yeah, I thought so too, but they both use Azure Service Bus, and running them parallel would make it more fragile. I change it like this, so there wasn't any overloap anymore.

Reply:
Can't we run them against different queues then? 🤔 But it's ok, it will only highly increase our CI time

Reply:
Yes, that would be better indeed. I just didn't want to get into that in this PR 😅 , I'll create an issue for this.

Reply:
https://github.com/arcus-azure/arcus.messaging/issues/121